compute/zgbmm.c

```diff
@@ -0,0 +1,410 @@
+ *          otherwise, ldb >= max(1,n).
+ *
+ * @param[in] beta
+ *          The scalar beta.
+ *
+ * @param[in,out] pC
+ *          An ldc-by-n matrix. On exit, the array is overwritten by the m-by-n
+ *          matrix ( alpha*op( A )*op( B ) + beta*C ).
+ *
+ * @param[in] ldc
+ *          The leading dimension of the array C. ldc >= max(1,m).
+ *
+ *******************************************************************************
+ *
+ * @retval PlasmaSuccess successful exit
+ *
+ *******************************************************************************
+ *
+ * @sa plasma_omp_zgbmm
+ * @sa plasma_cgbmm
+ * @sa plasma_dgbmm
+ * @sa plasma_sgbmm
+ *
+ ******************************************************************************/
+int plasma_zgbmm(plasma_enum_t transa, plasma_enum_t transb,
+                 int m, int n, int k, int kl, int ku,
+                 plasma_complex64_t alpha, plasma_complex64_t *pA, int lda,
+                                           plasma_complex64_t *pB, int ldb,
+                 plasma_complex64_t beta,  plasma_complex64_t *pC, int ldc)
+{
+    // Get PLASMA context.
+    plasma_context_t *plasma = plasma_context_self();
+    if (plasma == NULL) {
+        plasma_error("PLASMA not initialized");
+        return PlasmaErrorNotInitialized;
+    }
+
+    // Check input arguments.
+    if ((transa != PlasmaNoTrans) &&
+        (transa != PlasmaTrans) &&
+        (transa != PlasmaConjTrans)) {
+        plasma_error("illegal value of transa");
+        return -1;
+    }
+    if ((transb != PlasmaNoTrans) &&
+        (transb != PlasmaTrans) &&
+        (transb != PlasmaConjTrans)) {
+        plasma_error("illegal value of transb");
+        return -2;
+    }
+    if (m < 0) {
+        plasma_error("illegal value of m");
+        return -3;
+    }
+    if (n < 0) {
+        plasma_error("illegal value of n");
+        return -4;
+    }
+    if (k < 0) {
+        plasma_error("illegal value of k");
+        return -5;
+    }
+    if (kl < 0) {
+        plasma_error("illegal value of kl");
+        return -6;
+    }
+    if (ku < 0) {
+        plasma_error("illegal value of ku");
+        return -7;
+    }
+
+    int am, an;
+    int bm, bn;
+    if (transa == PlasmaNoTrans) {
+        am = m;
+        an = k;
+    }
+    else {
+        am = k;
+        an = m;
+    }
+    if (transb == PlasmaNoTrans) {
+        bm = k;
+        bn = n;
+    }
+    else {
+        bm = n;
+        bn = k;
+    }
+
+    if (lda < imax(1, am)) {
+        plasma_error("illegal value of lda");
+        return -8;
+    }
+    if (ldb < imax(1, bm)) {
+        plasma_error("illegal value of ldb");
+        return -10;
+    }
+    if (ldc < imax(1, m)) {
+        plasma_error("illegal value of ldc");
+        return -13;
+    }
+
+    // quick return
+    if (m == 0 || n == 0 || ((alpha == 0.0 || k == 0) && beta == 1.0))
+        return PlasmaSuccess;
+    // Tune parameters.
+    plasma->tuning = PlasmaEnabled;
```

Comment:
(From previous PR): You are right: plasma-> tuning line should be removed. (197)